### ISSUE TITLE
Bug #1: Rate Limiter

### ISSUE THREAD

**Issue by BadDub** [glitch100/Halo-API#1]
https://github.com/glitch100/Halo-API/blob/master/HaloEzAPI/Limits/RequestRateHttpClient.cs#L85

If Stopwatch.Elapsed.Seconds >= SecondsLimit than the following equation '(SecondsLimit - seconds) \* 1000' results in a negative number. i.e. (10 - 11) \* 1000 = -1000.
You can't sleep for a negative amount of time.

**Comment by prom3theu5** [comment]
Yes - this is definitely correct.

For now, work around with: 

```
int seconds = (SecondsLimit - Stopwatch.Elapsed.Seconds) * 1000;
int sleep = seconds > 0 ? seconds : seconds * -1;
Thread.Sleep(sleep);
```

**Comment by glitch100** [comment]
@BadDub  @prom3theu5 

Thanks both for noticing this. Amateur hour over here. I have implemented that fix and it is available in 1.0.3.
